manifests/params.pp

```diff
@@ -64,8 +64,8 @@
   $foreman_service_ensure = 'running'
   $foreman_service_enable = true
   $foreman_service_puma_threads_min = undef
-  $foreman_service_puma_threads_max = 16
-  $foreman_service_puma_workers = 2
+  $foreman_service_puma_threads_max = 5
+  $foreman_service_puma_workers = $facts['processors']['count'] * 1.5
```

Comment:
Should we have some upper bound as well? In the past we've seen that too many workers could also easily consume way too much memory.

Reply:
```suggestion
  $foreman_service_puma_workers = ($facts['processors']['count'] * 3) / 2
```

I think this works better since `$facts['processors']['count']` isn't guaranteed to be an even number -- there are a few physical CPUs that have an odd number of cores, or a VM could be created which has an arbitrary number of CPU cores.

This way we guarantee the value of `$foreman_service_puma_workers` remains an integer.

Reply:
Good point, but I'd prefer to use [floor](https://puppet.com/docs/puppet/6/function.html#floor) then. `1.5` is easier to read than `* 3 / 2`. Especially since it depends on the language how integer division happens (for example, Python 2 and Python 3 actually have different behavior).
```suggestion
  $foreman_service_puma_workers = floor($facts['processors']['count'] * 1.5)
```